### PR TITLE
Feature/ Forgot password

### PR DESCRIPTION
**Pull request by Mekhak11**

### CODE REVIEW

**Comment by Mekhak11** [comment]
https://github.com/user-attachments/assets/417db0ba-1dcb-48dc-9439-90c65ccc6344

**Comment by Mekhak11** [comment]
https://github.com/user-attachments/assets/ab667fd2-bca9-4113-8dcd-ce0fdcc64290

File: Birthday/Birthday/Modules/Auth/ResetPassword/ViewModel/ResetPasswordViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    )
+    forgotPasswordRepository.confirmCode(payload: payload)
+      .sink { [weak self] result in
+        self?.isLoading = false
+        switch result {
+        case .failure(let error):
+          Console.log(error)
+        default: break
+        }
+      } receiveValue: { isChanged in
```
Comment (nellyAslanyan) [inline Birthday/Birthday/Modules/Auth/ResetPassword/ViewModel/ResetPasswordViewModel.swift]: weak self in missing

File: Birthday/Birthday/Modules/Auth/ResetPassword/ViewModel/ResetPasswordViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  @Published var isPasswordValid: Bool = false
+  @Published var isConfirmPassValid: Bool = false
+  @Published var password: String = ""
+  @Published var confirmPassword: String = ""
+  
+  private let forgotPasswordRepository: ForgotPasswordRepository
+  private let passwordCode: String
+  private var cancelables = Set<AnyCancellable>()
+
+
```
Comment (nellyAslanyan) [inline Birthday/Birthday/Modules/Auth/ResetPassword/ViewModel/ResetPasswordViewModel.swift]: extra line

File: Birthday/Birthday/Modules/Auth/FrogetPassword/ViewModel/ForgotPasswordViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  private let forgotPasswordRepository: ForgotPasswordRepository
+  private var cancelables = Set<AnyCancellable>()
+  
+  init(forgotPasswordRepository: ForgotPasswordRepository) {
+    self.forgotPasswordRepository = forgotPasswordRepository
+    setupCodeValidation()
+    setupEmailValidation()
+  }
+  
+  func askForCode() {
```
Comment (nellyAslanyan) [inline Birthday/Birthday/Modules/Auth/FrogetPassword/ViewModel/ForgotPasswordViewModel.swift]: lets rename getCode :D

File: Birthday/Birthday/Modules/Auth/ResetPassword/ViewModel/ResetPasswordViewModel.swift (hunk 1 above)
Comment (Mekhak11) [reply Birthday/Birthday/Modules/Auth/ResetPassword/ViewModel/ResetPasswordViewModel.swift]: but we don't use self here.

File: Birthday/Birthday/Modules/Auth/FrogetPassword/ViewModel/ForgotPasswordViewModel.swift (hunk 3 above)
Comment (Mekhak11) [reply Birthday/Birthday/Modules/Auth/FrogetPassword/ViewModel/ForgotPasswordViewModel.swift]: fixed.

File: Birthday/Birthday/Modules/Auth/ResetPassword/ViewModel/ResetPasswordViewModel.swift (hunk 2 above)
Comment (Mekhak11) [reply Birthday/Birthday/Modules/Auth/ResetPassword/ViewModel/ResetPasswordViewModel.swift]: fixed.

**Comment by Mekhak11** [comment]
https://github.com/user-attachments/assets/5b6fd0bb-6e55-4325-b44f-5a1ace936452

File: Birthday/Birthday/Modules/Auth/FrogetPassword/View/ForgotPasswordScreen.swift [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+import SwiftUI
+
+struct ForgotPasswordScreen<T: ForgotPasswordViewModeling>: View {
+  
+  @StateObject var viewModel: T
+  @EnvironmentObject var router: NavigationRouter
+  @EnvironmentObject var appState: AppState
+  
+
```
Comment (nellyAslanyan) [inline Birthday/Birthday/Modules/Auth/FrogetPassword/View/ForgotPasswordScreen.swift]: extra line

Comment (Mekhak11) [reply Birthday/Birthday/Modules/Auth/FrogetPassword/View/ForgotPasswordScreen.swift]: fixed.

**Review by nellyAslanyan — APPROVED** [review]

**Pull request merged**